### PR TITLE
build(table): explicitly define MatTable ctor

### LINKED ISSUE #9329: [Table] Does not work when targetting ES6
#### Bug, feature request, or proposal:
Bug

#### What is the expected behavior?
Table appears in browser.

#### What is the current behavior?
Table doesn't appear.  Error message in console:

```
ERROR TypeError: Cannot read property 'nativeElement' of undefined
    at new CdkTable (table.js:438)
    at new MatTable (table.js:26)
    at createClass (core.js:10365)
    at createDirectiveInstance (core.js:10212)
    at createViewNodes (core.js:11672)
    at callViewAction (core.js:12106)
    at execComponentViewsAction (core.js:12015)
    at createViewNodes (core.js:11700)
    at createRootView (core.js:11561)
    at callWithDebugContext (core.js:12927)

```

#### What are the steps to reproduce?

Create a new angular 5 project.  Add the example code from material.angular.io for mat-table.

It should run fine.

Now go into tsconfig.json and target something higher than es5.  Boom.

#### What is the use-case or motivation for changing an existing behavior?
Projects targeting > es5.

#### Which versions of Angular, Material, OS, TypeScript, browsers are affected?
Angular 5+, Material 5-rc0 - 5.0.4, TS 2.4.2
Windows 7 64bit
Chrome

### PR DESCRIPTION
**Pull request by andrewseguin**
Explicitly define the ctor due to a tsickle issue which is depended on by compiler-cli.

Fixes #9329

### CODE REVIEW

File: src/lib/table/table.ts [hunk 1]
```diff
@@ -24,4 +32,15 @@ import {CDK_TABLE_TEMPLATE, CdkTable} from '@angular/cdk/table';
   encapsulation: ViewEncapsulation.None,
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class MatTable<T> extends CdkTable<T> { }
+export class MatTable<T> extends CdkTable<T> {
+  // TODO(andrewseguin): Remove this explicitly set ctor when the compiler knows how to properly
```
Comment (jelbourn) [inline src/lib/table/table.ts]: Prefer `constructor` to `ctor`

**Comment by jelbourn** [comment]
Marking as P1, then.

**Comment by VangelisAlSghir** [comment]
Can we expect this to be in RC.15? We really need to use Material Tables targeting ES6.

**Comment by andrewseguin** [comment]
@VangelisAlSghir Yes, it should be fixed as of `6.0.0-rc.14` (released a few days ago)

**Comment by VangelisAlSghir** [comment]
Are you sure? I just made a new angular project and targeted ES6 and tried to use a material table with the example code from https://github.com/angular/material2/blob/6.0.0-rc.14/src/demo-app/table/mat-table-data-source/mat-table-data-source.html but I still get following errors in the console:

TypeError: Cannot read property 'find' of undefined
    at MatHeaderRowDef.ngOnChanges (webpack-internal:///./node_modules/@angular/cdk/esm2015/table.js:76)

and

ypeError: Cannot read property 'diff' of undefined
    at MatRowDef.getColumnsDiff (webpack-internal:///./node_modules/@angular/cdk/esm2015/table.js:86)

**Comment by andrewseguin** [comment]
Ah bummer, looks like the same "fix" needs to be applied to those classes as well. Will be able to make a change next week but it'll like be a part of v6.0.1 or v6.0.2

**Comment by VangelisAlSghir** [comment]
The sooner the better for us :)

**Comment by andrewseguin** [comment]
PR has been created, hoping it'll be merged this week and go out with our next release on Monday. Thanks for your patience and persistence with the issue

**Pull request merged**